File: server/server/src/main/java/ca/mcgill/splendorserver/models/GameConfigData.java
package ca.mcgill.splendorserver.models;


/**
 * Game config data.
 * 
 */
public class GameConfigData {
  
  private String gameName;
  private String hostId;
  private String[] playerIds;
  
  private CardData[] deck1;
  private CardData[] deck2;
  private CardData[] deck3;
  private CardData[] exDeck1;
  private CardData[] exDeck2;
  private CardData[] exDeck3;
  
  private NobleData[] allNobles;
  
  /**
   * Gets game name.
   *
   * @return gameName
   */
  public String getGameName() {
    return gameName;
  }
  
  /**
   * Gets host ID.
   *
   * @return hostId
   */
  public String getHostId() {
    return hostId;
  }
  
  /**
   * Gets player IDs.
   *
   * @return hostId
   */
  public String[] getPlayerIds() {
    return playerIds;
  }
  
  /**
   * Gets level 1 deck.
   *
   * @return deck1
   */
  public CardData[] getDeck1() {
    return deck1;
  }
  
  /**
   * Gets level 2 deck.
   *
   * @return deck2
   */
  public CardData[] getDeck2() {
    return deck2;
  }
  
  /**
   * Gets level 3 deck.
   *
   * @return deck3
   */
  public CardData[] getDeck3() {
    return deck3;
  }
  
  /**
   * Gets level 1 expansion deck.
   *
   * @return expansion deck1
   */
  public CardData[] getExDeck1() {
    return exDeck1;
  }
  
  /**
   * Gets level 2 expansion deck.
   *
   * @return expansion deck2
   */
  public CardData[] getExDeck2() {
    return exDeck2;
  }
  
  /**
   * Gets level 3 expansion deck.
   *
   * @return expansion deck13
   */
  public CardData[] getExDeck3() {
    return exDeck3;
  }
  
  /**
   * Gets all nobles available.
   *
   * @return nobles
   */
  public NobleData[] getAllNobles() {
    return allNobles;
  }
}

File: client/Splendor Client/Assets/Scripts/DataManagement/GameConfigData.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[System.Serializable]
public class GameConfigData {
    public string gameName;
    public string hostId;
    public string[] playerIds;
  
    public CardData[] deck1;
    public CardData[] deck2;
    public CardData[] deck3;
    public CardData[] exDeck1;
    public CardData[] exDeck2;
    public CardData[] exDeck3;
    
    public NobleData[] allNobles;

    public GameConfigData() { }
    public GameConfigData(Authentication mainPlayer, Session session, AllCards allCards, NobleRow allNobles) {
        gameName = session.GetVariant();
        hostId = mainPlayer.username;
        playerIds = new string[4];
        for(int i = 0;i < session.players.Count; i++) {
            playerIds[i] = session.players[i];
        }
        deck1 = allCards.cards[0].deck.ToArray();
        deck2 = allCards.cards[1].deck.ToArray();
        deck3 = allCards.cards[2].deck.ToArray();
        exDeck1 = allCards.cards[3].deck.ToArray();
        exDeck2 = allCards.cards[4].deck.ToArray();
        exDeck3 = allCards.cards[5].deck.ToArray();

        this.allNobles = allNobles.ToArray();
    }


}
